### PR TITLE
add check-scalar

### PR DESCRIPTION
**Pull request by fullflu**
# Overview

- We use the `check_scalar` function to validate scalar (int or float) variables
- Fix the format of error messages as that of `skearn.utils.check_scalar`

# Change point

- [x] OPE modules
- [x] Dateset modules
- [x] Policy modules
- [x] OPE test
- [x] Dataset test
- [x] Polucy test
- [x] utils

# Remark

- Although we can check the condition of `a >= 0` easily by the `check_scalar` function,  the condition of `a > 0` cannot be checked easily by the function. Thus, we manually check the condition.

### CODE REVIEW

File: obp/policy/linear.py [hunk 1]
```diff
@@ -145,10 +144,9 @@ def select_action(self, context: np.ndarray) -> np.ndarray:
             List of selected actions.
 
         """
-        if context.ndim != 2 or context.shape[0] != 1:
-            raise ValueError(
-                f"context shape must be (1, dim_context),but {context.shape} is given"
-            )
+        check_array(array=context, name="context", expected_dim=2)
+        if context.shape[0] != 1:
+            raise ValueError("Expected `context.shape[1] == 1`, but found it False")
```
Comment (usaito) [inline obp/policy/linear.py:149]: "Expected `context.shape[0] == 1`, but found it False" ?

File: obp/policy/linear.py [hunk 2]
```diff
@@ -225,10 +220,10 @@ def select_action(self, context: np.ndarray) -> np.ndarray:
...
```
Comment (usaito) [inline obp/policy/linear.py:225]: "Expected context.shape[0] == 1, but found it False" ?

**Review by usaito — COMMENTED** [review]
@fullflu Thanks! Can you fix some tiny points?

File: obp/policy/logistic.py [hunk 3]
```diff
@@ -49,15 +49,13 @@ class BaseLogisticPolicy(BaseContextualPolicy):
...
+        check_scalar(self.alpha_, "alpha_", float)
+        if self.alpha_ <= 0.0:
+            raise ValueError(f"`alpha_`= {self.alpha_}, must be > 0.0.")
 
-        if not isinstance(self.lambda_, float) or self.lambda_ <= 0.0:
-            raise ValueError(
-                f"lambda_ should be a positive float, but {self.lambda_} is given"
-            )
+        check_scalar(self.lambda_, "lambda_", float)
+        if self.alpha_ <= 0.0:
```
[the remark this answers is not in the document]
Comment (fullflu) [reply obp/policy/logistic.py:57]: I will add several tests about those arguments...!

**Pull request merged**